### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by johnqa** [cloud-bulldozer/kube-burner#219]
**Describe the bug**
`panic: runtime error: invalid memory address or nil pointer dereference)` error when running kube-burner with api-testing (but also other tests) on a Rancher cluster. 

I see the error appears after the results are reported,
Moreover, the resources created (deployments, secretes, services) are not removed.

Please let me know if more information is needed.

**To Reproduce**
I run kube-burner using docker image:
```docker run --rm -e KUBECONFIG=/tmp/burner/config -v /api-intensive:/tmp/burner cloud-bulldozer/kube-burner init --config /tmp/burner/api-intensive.yml --uuid uuidgen```

**Results**
```time="2022-12-01 14:06:16" level=info msg="api-intensive: PodScheduled 50th: 164 99th: 1108 max: 2536 avg: 212"
time="2022-12-01 14:06:16" level=info msg="api-intensive: ContainersReady 50th: 30943 99th: 156269 max: 159664 avg: 60022"
time="2022-12-01 14:06:16" level=info msg="api-intensive: Initialized 50th: 2675 99th: 73400 max: 74635 avg: 16586"
time="2022-12-01 14:06:16" level=info msg="api-intensive: Ready 50th: 30943 99th: 156269 max: 159664 avg: 60022"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x110 pc=0x170cbad]

goroutine 1 [running]:
github.com/cloud-bulldozer/kube-burner/pkg/burner.Run({{{{0xc000277704, 0x7}, {0xc0002453c0, 0x1, 0x1}, {0xc0002776d0, 0xb}, 0x0, 0x1, 0x0}, ...}, ...}, ...)
	/home/runner/work/kube-burner/kube-burner/pkg/burner/job.go:143 +0xf6d
main.initCmd.func1(0xc00038c000?, {0xc000500fc0?, 0x4?, 0x4?})
	/home/runner/work/kube-burner/kube-burner/cmd/kube-burner/kube-burner.go:128 +0x726
github.com/spf13/cobra.(*Command).execute(0xc00038c000, {0xc000500f80, 0x4, 0x4})
	/home/runner/work/kube-burner/kube-burner/vendor/github.com/spf13/cobra/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x2bd3280)
	/home/runner/work/kube-burner/kube-burner/vendor/github.com/spf13/cobra/command.go:1044 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/kube-burner/kube-burner/vendor/github.com/spf13/cobra/command.go:968
main.main()
	/home/runner/work/kube-burner/kube-burner/cmd/kube-burner/kube-burner.go:349 +0x2b2
```

**Comment by rsevilla87** [comment]
Hey @johnqa!, thanks for reporting this, the resources are not removed, that's expected, as that feature doesn't exist at the moment. I have plans to implement it asap though.

With respect to the segmentation fault issue, I'm digging into it, give me some time. Thanks!

In the meantime, can you send me the output of?

docker run --rm quay.io/cloud-bulldozer/kube-burner:latest version

**Comment by rsevilla87** [comment]
@johnqa The previous PR fixed the issue in my env, can you try again?

**Comment by johnqa** [comment]
Thank you for your quick answer.

This is my output:
```
Version: latest
Git Commit: dca1acdbf75a73e1abad16af6953b3d416d655ea
Build Date: 2022-12-02-10:48:07
Go Version: go1.18.8
OS/Arch: linux amd64

time="2022-12-02 13:41:20" level=info msg="🔥 Starting kube-burner (latest@dca1acdbf75a73e1abad16af6953b3d416d655ea) with UUID c907a072-53d4-43ca-8cfc-03e1e5797a63"
time="2022-12-02 13:41:20" level=info msg="🔥 Starting kube-burner (latest@dca1acdbf75a73e1abad16af6953b3d416d655ea) with UUID c907a072-53d4-43ca-8cfc-03e1e5797a63"
time="2022-12-02 13:41:20" level=info msg="📈 Creating measurement factory"
time="2022-12-02 13:41:20" level=info msg="Registered measurement: podLatency"
time="2022-12-02 13:41:20" level=info msg="Job api-intensive: 10 iterations with 1 Deployment replicas"
time="2022-12-02 13:41:20" level=info msg="Job api-intensive: 10 iterations with 1 ConfigMap replicas"
time="2022-12-02 13:41:20" level=info msg="Job api-intensive: 10 iterations with 1 Secret replicas"
time="2022-12-02 13:41:20" level=info msg="Job api-intensive: 10 iterations with 1 Service replicas"
time="2022-12-02 13:41:20" level=info msg="Pre-load: images from job api-intensive"
......
time="2022-12-02 13:43:26" level=info msg="api-intensive: Ready 50th: 12598 99th: 16928 max: 16928 avg: 13436"
time="2022-12-02 13:43:26" level=info msg="api-intensive: PodScheduled 50th: 75 99th: 271 max: 271 avg: 94"
time="2022-12-02 13:43:26" level=info msg="api-intensive: ContainersReady 50th: 12598 99th: 16928 max: 16928 avg: 13436"
time="2022-12-02 13:43:26" level=info msg="api-intensive: Initialized 50th: 510 99th: 2314 max: 2314 avg: 817"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x110 pc=0x170cbad]

goroutine 1 [running]:
github.com/cloud-bulldozer/kube-burner/pkg/burner.Run({{{{0xc000486164, 0x7}, {0xc0004ef7d0, 0x1, 0x1}, {0xc000486140, 0xb}, 0x0, 0x1, 0x0}, ...}, ...}, ...)
	/home/runner/work/kube-burner/kube-burner/pkg/burner/job.go:143 +0xf6d
main.initCmd.func1(0xc000546600?, {0xc00071c200?, 0x4?, 0x4?})
	/home/runner/work/kube-burner/kube-burner/cmd/kube-burner/kube-burner.go:128 +0x726
github.com/spf13/cobra.(*Command).execute(0xc000546600, {0xc00071c1c0, 0x4, 0x4})
	/home/runner/work/kube-burner/kube-burner/vendor/github.com/spf13/cobra/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x2bd5280)
	/home/runner/work/kube-burner/kube-burner/vendor/github.com/spf13/cobra/command.go:1044 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/kube-burner/kube-burner/vendor/github.com/spf13/cobra/command.go:968
main.main()
	/home/runner/work/kube-burner/kube-burner/cmd/kube-burner/kube-burner.go:349 +0x2b2

```

**Comment by rsevilla87** [comment]
> dca1acdbf75a73e1abad16af6953b3d416d655ea

According to that you're using a kube-burner version based on https://github.com/cloud-bulldozer/kube-burner/commit/dca1acdbf75a73e1abad16af6953b3d416d655ea, the fix was introduced in aedb81893234d97ec982457af57dac8478e759f6, can you ensure you're using the latest image available?

**Comment by johnqa** [comment]
Indeed that was the problem. The issue is fixed with the latest image.

Thank you for the fast fix.
